flink-streaming-java/src/main/java/org/apache/flink/streaming/api/operators/async/AsyncWaitOperator.java

```diff
@@ -402,8 +402,10 @@ private void stopResources(boolean waitForShutdown) throws InterruptedException
 		pendingStreamElementQueueEntry = streamElementQueueEntry;
 
 		while (!queue.tryPut(streamElementQueueEntry)) {
+			yieldToDownstream();
```

Comment:
What do you think about having separate PRs for introducing `yieldToDownstream()` implementation and it's usage in `AyncWaitOperator`?

Reply:
Yes, I can do that. But would first like to have a general feedback on the design.
Btw the commit order has been shuffled by github for some reasons. This is actually the last commit.